replications/internal/queue_management_test.go

```diff
@@ -30,6 +32,81 @@ func TestCreateNewQueueDirExists(t *testing.T) {
 	require.DirExists(t, filepath.Join(queuePath, id1.String()))
 }
 
+func TestEnqueueScanLog(t *testing.T) {
+	t.Parallel()
+
+	// Initialize queue manager with zap observer (to allow assertions on log messages)
+	enginePath, err := os.MkdirTemp("", "engine")
+	require.NoError(t, err)
+	queuePath := filepath.Join(enginePath, "replicationq")
+
+	observedZapCore, observedLogs := observer.New(zap.InfoLevel)
+	observedLogger := zap.New(observedZapCore)
+
+	qm := NewDurableQueueManager(observedLogger, queuePath)
+	defer os.RemoveAll(filepath.Dir(queuePath))
+
+	// Create new queue
+	err = qm.InitializeQueue(id1, maxQueueSizeBytes)
+	require.NoError(t, err)
+
+	// Enqueue some data
+	testData := "weather,location=us-midwest temperature=82 1465839830100400200"
+	err = qm.EnqueueData(id1, []byte(testData))
+	require.NoError(t, err)
+
+	// Give it a second to scan the queue
+	time.Sleep(time.Second)
+
+	// Check that data the scanner logs is the same as what was enqueued
+	require.Equal(t, 1, observedLogs.Len())
```

Comment:
Hmm I don't love asserting the results of log messages for this, especially since the callback to `rq.SendWrite` is not in its final form.

Are we specifically trying to test the logging output in this test?

Reply:
Not specifically trying to test log output, but the idea was to assert on logs for now (to check that data is the same after enqueueing and scanning), then modify those tests after the remote write part is implemented. So rather than a log message, it would then be checking to see if the remote write request contains the original data. I'm not sure if there's a different way we could approach it for now, or just remove the tests and add something back in once the remote write is testable?

Reply:
Per a separate discussion, we decided that it would make sense to treat the "write" function as a dependency so that we can inject versions of it specialized for testing. For now, the tests which assert based on log messages server as a proxy for that and provide some confidence that the changes in this PR are doing what they are supposed to be doing. A follow-up PR will include the "write" function dependency injection and refactor the tests accordingly.